src/explore-education-statistics-frontend/src/modules/table-tool/components/__tests__/TableToolShare.test.tsx

```diff
@@ -52,31 +52,78 @@ describe('TableToolShare', () => {
     );
 
     await waitFor(() => {
-      expect(
-        screen.queryByRole('button', {
-          name: 'Generate shareable link',
-        }),
-      ).not.toBeInTheDocument();
+      expect(screen.getByText('Generated share link')).toBeInTheDocument();
+    });
+
+    expect(
+      screen.queryByRole('button', {
+        name: 'Generate shareable link',
+      }),
+    ).not.toBeInTheDocument();
+
+    const urlInput = screen.getByTestId('permalink-generated-url');
+    expect(urlInput).toBeInTheDocument();
+    expect(urlInput).toHaveValue(
+      'http://localhost/data-tables/permalink/permalink-id',
+    );
+
+    expect(
+      screen.getByRole('button', {
+        name: 'Copy link',
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.getByRole('link', {
+        name: 'View share link',
+      }),
+    ).toBeInTheDocument();
+  });
+
+  test('copies the link to the clipboard when the copy button is clicked', async () => {
+    permalinkService.createPermalink.mockResolvedValue({
+      id: 'permalink-id',
+    } as Permalink);
+
+    const originalClipboard = { ...window.navigator.clipboard };
+    Object.assign(window.navigator, {
+      clipboard: {
+        writeText: jest.fn(),
+      },
+    });
```

Comment:
Would suggest doing all this mocking setup in common's `setupGlobals.js`:

```js
beforeAll(() => {
  window.navigator.clipboard = {
    writeText: jest.fn(),
  };
});
```

Note that we don't shouldn't need to mess around with `Object.assign` here

Reply:
Good call, thanks.